packages/tldraw/src/lib/ui/components/Toolbar/OverflowingToolbar.tsx

```diff
@@ -95,44 +122,24 @@ export function OverflowingToolbar({ children }: OverflowingToolbarProps) {
 	}, [onDomUpdate])
 
 	useEffect(() => {
-		const keys = [
-			['1', 0],
-			['2', 1],
-			['3', 2],
-			['4', 3],
-			['5', 4],
-			['6', 5],
-			['7', 6],
-			['8', 7],
-			['9', 8],
-			['0', 9],
-		] as const
-
-		for (const [key, index] of keys) {
-			hotkeys(key, (event) => {
-				if (areShortcutsDisabled(editor)) return
-				preventDefault(event)
-
-				const relevantEls = Array.from(mainToolsRef.current?.children ?? []).filter(
-					(el): el is HTMLElement => {
-						// only count html elements...
-						if (!(el instanceof HTMLElement)) return false
-
-						// ...that are buttons...
-						if (el.tagName.toLowerCase() !== 'button') return false
+		if (!editor.options.enableToolbarKeyboardShortcuts) return
 
-						// ...that are actually visible
-						return !!(el.offsetWidth || el.offsetHeight)
-					}
-				)
-
-				const el = relevantEls[index]
-				if (el) el.click()
-			})
+		function handleKeyDown(event: KeyboardEvent) {
+			if (areShortcutsDisabled(editor)) return
+			// no accelerator keys
+			if (event.ctrlKey || event.metaKey || event.altKey || event.shiftKey) return
+			for (const [key, index] of KEYS) {
```

Comment:
if we make KEYS an object instead of an array we could skip this iteration on every keypress and instead just look up the relevant index